crates/solver/src/settlement_ranker.rs

```diff
@@ -208,6 +210,32 @@ impl SettlementRanker {
             });
         }
 
+        // Filter out settlements with too high score.
+        if Utc::now() > self.auction_rewards_activation_timestamp {
+            // CIP20 activated
+            rated_settlements.retain(|(solver, settlement, _)| {
+                let surplus = big_rational_to_u256(&settlement.surplus).unwrap_or(U256::MAX);
+                let fees = big_rational_to_u256(&settlement.solver_fees).unwrap_or(U256::MAX);
+                let max_score = surplus.saturating_add(fees);
+                let valid_score = settlement.score.score() < max_score;
```

Comment:
The max score itself is also valid, no?
```suggestion
                let valid_score = settlement.score.score() <= max_score;
```